### PR TITLE
Update clustering.md to be more current

### PR DESCRIPTION
**Pull request by fosterseth**
##### SUMMARY

these docs were wildly out of date, and also written as a proposal / ADR style

- Update to reflect k8s deployment type (as opposed to older docker style deployment pre-18.0)
- Describe communications between nodes in a clustered environment

##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - Docs

### CODE REVIEW

File: docs/clustering.md [hunk 1]
```diff
@@ -1,92 +1,168 @@
...
+               └───────────┘
+```
+
+There are two main deployment types, virtual machines (VM) or K8S. Ansible Automation Controller installation supports both VM or K8S deployments. The upstream AWX project only supports K8S deployments. Either deployment type supports cluster scaling.
+- Control plane nodes run a number of background services that are managed by supervisord
+  - dispatcher
+  - wsbroadcast
+  - callback receiver
+  - receptor (*managed under systemd)
+  - redis (*managed under systemd)
```
Comment (AlanCoding) [inline docs/clustering.md:42]: Thank you for writing these out!

File: docs/clustering.md [hunk 2]
```diff
@@ -1,92 +1,168 @@
...
+  - uwsgi
+  - daphne
+  - rsyslog
+- For K8S deployments, these background processes are containerized
+  - `awx-ee`: receptor
+  - `awx-web`: uwsgi, daphne, wsbroadcast, rsyslog
+  - `awx-task`: dispatcher, callback receiver
+  - `redis`: redis
+- Each control node is monolithic and contains all the necessary components for handling API requests and running jobs.
```
Comment (AlanCoding) [inline docs/clustering.md:51]: for now... will change soon. But maybe this is a good strategy, as your updated wording makes it very clear where the new web/task split feature needs to update.

**Comment by AlanCoding** [comment]
:+1: this content looks great, I just want to ask @shanemcd to help us get the proper doc scope regarding AAP, perhaps we should link to AAP installer docs. So explaining its operation isn't really a goal here, even though mentions are already fairly minimal.

File: docs/clustering.md [hunk 3]
```diff
@@ -1,92 +1,168 @@
...
-  * Job Templates
+```
+       ┌───────────────────────────┐
+       │      Load-balancer        │
+       │   (configured separately) │
+       └───┬───────────────────┬───┘
+           │   round robin API │
+           ▼       requests    ▼
+
+  AWX VM/Pod 1               AWX VM/Pod 2
```
Comment (shanemcd) [inline docs/clustering.md]: @fosterseth This still says "AWX VM".

File: docs/clustering.md [hunk 4]
```diff
@@ -1,92 +1,168 @@
...
+## Scaling the cluster
+
+For VM (AAP) deployments, scaling up involves modifying `inventory`and re-running setup.sh
+For K8s deployments, scaling up is handled by changing the number of replicas in the AWX replica set.
+
+After scaling up, the new control plane node is registered in the database as a new `Instance`.
 
-  ...such that execution of jobs under those resources will favor particular queues.
+Instance types:
+`hybrid` - control plane node that can also run jobs
```
Comment (shanemcd) [inline docs/clustering.md]: Please mention that hybrid and hop nodes are not available in default installations of AWX.

File: docs/clustering.md (hunk 3 above)
Comment (fosterseth) [reply docs/clustering.md]: I updated to be more generic `AWX Control Node`

File: docs/clustering.md (hunk 4 above)
Comment (fosterseth) [reply docs/clustering.md]: I made it clear these are part of AAP only

**Review by AlanCoding — APPROVED** [review]
way way better than what we had before. I think we should merge and then iterate on it from there as feature work continues.

**Comment by gundalow** [comment]
> way way better than what we had before. I think we should merge and then iterate on it from there as feature work continues.

Agreed, this is a solid improvement on what we currently have.

**Pull request merged**